android/src/main/java/com/adyenreactnativesdk/component/dropin/AdyenDropInComponent.kt

```diff
@@ -200,5 +224,64 @@ class AdyenDropInComponent(context: ReactApplicationContext?) : BaseModule(conte
     companion object {
         private const val TAG = "DropInComponent"
         private const val COMPONENT_NAME = "AdyenDropIn"
+        private var dropInLauncher: ActivityResultLauncher<Intent>? = null
+        private val dropInCallback = DropInCallbackListener()
+
+        @JvmStatic
+        fun setDropInLauncher(activity: ActivityResultCaller) {
+            dropInLauncher = activity.registerForActivityResult(ReactDropInResultContract(), dropInCallback::onDropInResult)
+        }
+
+        @JvmStatic
+        fun removeDropInLauncher() {
+            dropInLauncher = null
+        }
     }
+
 }
+
+internal interface ReactDropInCallback {
+    fun onCancel()
+    fun onError(reason: String?)
+    fun onCompleated(result: String)
+}
+
+private class ReactDropInResultContract : ActivityResultContract<Intent, DropInResult?>() {
+    override fun createIntent(context: Context, input: Intent): Intent {
+        return input
+    }
+
+    override fun parseResult(resultCode: Int, intent: Intent?): DropInResult? {
+        return DropIn.handleActivityResult(DropIn.DROP_IN_REQUEST_CODE, resultCode, intent)
+    }
+}
+
+private class DropInCallbackListener: DropInCallback {
+
+    var dropInCallback: ReactDropInCallback? = null
+
+    override fun onDropInResult(dropInResult: DropInResult?) {
+        if (dropInResult == null) return
+        when (dropInResult) {
+            is DropInResult.CancelledByUser -> dropInCallback?.onCancel()
+            is DropInResult.Error -> dropInCallback?.onError(dropInResult.reason)
+            is DropInResult.Finished -> dropInCallback?.onCompleated(dropInResult.result)
+        }
+    }
+}
+
+class DropInException(code: String, message: String, cause: Throwable? = null) : KnownException(code = code, errorMessage = message, cause) {
+    companion object {
+        val NO_MODULE_LISTENER = DropInException(
```

Comment:
same sealed class approach applies here as well.

Reply:
👍

Reply:
Fixed in [b32d205](https://github.com/Adyen/adyen-react-native/pull/52/commits/b32d205b75611e56ce710cb8fdd41f980d4cb88a)